### ISSUE TITLE
Binary stripping behavior

### ISSUE THREAD

**Issue by cstrouse** [skycocker/chromebrew#1311]
Some packages require that their binaries are not stripped or that some of them are not stripped. We need a way to specify that the entire package not be stripped or possibly to specify a list of files to not strip.

The `go` package has test data files such as `/usr/local/lib/go/src/runtime/pprof/testdata/test32be` which are 32-bit ELF files but should not be stripped.

Another example are `OCaml` binaries which are ELF but also contain OCaml bytecode. Stripping some of these files removes the embedded byte code which renders the binaries non-functional.

**Comment by uberhacker** [comment]
@cstrouse: There is a section in the crew command that has exceptions for compressed files.  You could add a few more exceptions.  It would be nice if we could make the logic a bit more elegant also if you want to try and make a go at fixing it up a bit.
```
430 def install_package (pkgdir)
431   Dir.chdir pkgdir do
432     FileUtils.mv 'dlist', CREW_CONFIG_PATH + "meta/#{@pkg.name}.directorylist"
433     FileUtils.mv 'filelist', CREW_CONFIG_PATH + "meta/#{@pkg.name}.filelist"
434 
435     # Strip libraries with -S
436     strip_find_files "find . -type f -name 'lib*.a' -print", "-S"
437     strip_find_files "find . -type f -name 'lib*.so*' -print", "-S"
438 
439     # Strip binaries but not compressed archives
440     strip_find_files "find . -type f ! -iname '*\.bz2' ! -iname '*\.gz' ! -iname '*\.lha' ! -iname '*\.rar' ! -iname '*\.tar' ! -iname '*\.tgz' ! -iname '*\.xz' ! -iname '*\.zip' -perm /111 -print | sed -    e '/lib.*\.a$/d' -e '/lib.*\.so/d'"
441 
442     system "tar cf - ./usr/* | (cd /; tar xp --keep-directory-symlink -f -)"
443   end
444 end
```

**Comment by cstrouse** [comment]
jfrog also has issues with stripping due to an unrecognized file format.

**Comment by JL2210** [comment]
There is a solution in the `arduino_ide.rb` package. This can be closed.

**Comment by uberhacker** [comment]
@JL2210: Maybe you can post a Q & A in the faq?

**Comment by JL2210** [comment]
I don't think that would be necessary. This is probably only going to be used in very few packages, such as `arduino_ide`, `guile`, `ocaml`, and `go`.

**Comment by saltedcoffii** [comment]
This is now possible with environment variables in the package.rb file.
